sage-ios/SAGE/SAGE/LoginView.swift

```diff
@@ -0,0 +1,236 @@
+//
+//  LoginView.swift
+//  SAGE
+//
+//  Created by Sameera Vemulapalli on 10/7/15.
+//  Copyright © 2015 Cal Blueprint. All rights reserved.
+
+import UIKit
+
+class LoginView: UIView {
+    
+    var loginEmailField: UITextField?
+    var loginPasswordField: UITextField?
+    var signUpLink: UIButton?
+    var sageLabel: UILabel?
+    var containerView: UIView?
+    var firstDivider: UIView?
+    var secondDivider: UIView?
+    var loginButton: UIButton?
+    var loginButtonText: UILabel?
+    
+    var movedUp: Bool = false
+    
+    required override init(frame: CGRect) {
+        let screenRect = UIScreen.mainScreen().bounds;
+        let screenWidth = screenRect.size.width;
+        let screenHeight = screenRect.size.height;
+        let newFrame = CGRectMake(0, 0, screenWidth, screenHeight)
+        super.init(frame: newFrame)
+        self.setUpViews()
+    }
+    
+    override func layoutSubviews() {
+        self.positionViews()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        self.setUpViews()
+    }
+    
+    func setUpViews () {
+        self.setUpBackground()
+        self.setUpGestureRecognizer()
+        self.setUpKeyboardNotifications()
+        self.setUpLoginItems()
+    }
+    
+    func positionViews() {
+        self.setUpLoginArea()
+    }
+    
+    func setUpLoginItems() {
+        self.containerView = UIView()
```

Comment:
you dont need all these init lines. instead of making your properties all optional and force unwrapping them everytime, you can just do something above like var containerView: UIView = UIView()